src/domain/file/command.ts

```diff
@@ -43,3 +45,18 @@ export type Command = {
   // Remove a file in memory by a given file id
   removeFile: (fileId: FileId) => IOEither<ResourceNotFoundError, void>
 }
+
+export type FileError = ResourceConflictError | ResourceNotFoundError
+
+export const ShowFileError: Show<FileError> = {
+  show: (error: FileError): string => {
+    switch (error._tag) {
+      case 'resource-conflict': {
+        return `Oops... You are trying either to store a file whose id already exists in memory or to store files with the same id... So we can't store these files with these ids: [${error.fileIds}]`
```

Comment:
I'd be inclined to make a more summary error message:

```suggestion
        return `Error ${error._tag}: Failed to store resource with conflicting IDs [${error.fileIds.join(', ')}].`
```

If you want to improve the message, it's up to the caller to do so, notably by providing a little more context according to the use case.

Reply:
Really much better 👍